api/queries_pr.go

```diff
@@ -639,7 +639,7 @@ func PullRequestForBranch(client *Client, repo ghrepo.Interface, baseBranch, hea
 	query := `
 	query PullRequestForBranch($owner: String!, $repo: String!, $headRefName: String!) {
 		repository(owner: $owner, name: $repo) {
-			pullRequests(headRefName: $headRefName, states: OPEN, first: 30) {
+			pullRequests(headRefName: $headRefName, first: 30) {
```

Comment:
This fixes an immediate problem, but causes another one.

PullRequestForBranch is called in two places:
1. find the PR for a branch, e.g. in `pr view`;
2. check if there are open PRs for a branch during `pr create`.

This change would break the feature described in (2).

I think the solution should be to have PullRequestForBranch accept an argument that dictates whether we're searching through open PRs or all PRs regardless of state. Also, when searching over all PRs, precedence should still be given to an open PR if one exists, even if a closed PR for the same branch exists and is chronologically newer.